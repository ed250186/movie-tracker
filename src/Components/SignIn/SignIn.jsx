import React, {Component} from 'react';
import { connect }  from 'react-redux';
import "../SignUp/SignUp.scss";
import { signInUser } from "../../actions/userActions";
import { fetchUserSignIn } from "../../apiCalls/apiCalls.js";
import { NavLink } from "react-router-dom";


class SignIn extends Component {
  constructor() {
    super();
    this.state = {
      email: "",
      password: ""
    };
  }

  handleLogin = async event => {
    event.preventDefault();
    const { email, password } = this.state;
    let signInUser = await fetchUserSignIn(email, password);
    this.props.signInUser(signInUser.data);
    this.resetInputs();
  };

  handleChange = event => {
    event.preventDefault();
    this.setState({ [event.target.name]: event.target.value });
  };

  resetInputs = () => {
    this.setState({ email: "", password: "" });
  };

  render() {
    return (
      <div className='sign-in'>
        <form onSubmit={this.handleLogin}>
          <NavLink to='/'>
            <img src="" alt="exit sign-in button"/>
          </NavLink>
          <div>
          <input 
            name='email'
            value={this.state.email}
            placeholder='Name'
            onChange={event => this.handleChange(event)}
          />
          <input 
            type='password'
            name='password'
            value={this.state.password}
            placeholder='Password'
            onChange={event => this.handleChange(event)}
          />
          <input 
            type="submit" 
            value='Sign In' 
            className='button'
          />
          <h2>{this.state.loginMessage}</h2>
          </div>
        </form>
      </div>
    )
  }
}

const mapStateToProps = state => ({
  signInUser: state.signInUser
});

const mapDispatchToProps = dispatch => ({
  signInUser: user => dispatch(signInUser(user))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SignIn);
