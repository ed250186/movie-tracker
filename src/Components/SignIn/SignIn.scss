.SignIn {
    main {
    display: flex;
    justify-content: center;

    form {
        color: white;
        margin: 10px;
    }

    input {
        padding: 10px;
        width: auto;
    }
    }
}