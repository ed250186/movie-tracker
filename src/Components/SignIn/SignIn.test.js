import React from 'react';
import {shallow} from 'enzyme';
import {SignIn} from './SignIn';

//54, 25, 60, 59

//lines ... 25,26,28,30,90

describe('SignIn', () => {
  let wrapper
  let mockEvent = { 
    preventDefault: jest.fn(),
    target: {
    name: 'email',
    value: 'nathan.froeh@gmail.com'
  }}
  beforeEach(() => {
    wrapper = shallow(<SignIn />);
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should reset email and password state', () => {
    wrapper.setState({email: 'hi', password: 'hi'})
    expect(wrapper.state('email')).toEqual('hi')
    expect(wrapper.state('password')).toEqual('hi')
    wrapper.instance().resetInputs()
    expect(wrapper.state('email')).toEqual('')
    expect(wrapper.state('password')).toEqual('')
  })

  it('should be called on email change', () => {
    wrapper.instance().handleChange = jest.fn()
    wrapper.find('.email').simulate('change')
    expect(wrapper.instance().handleChange).toHaveBeenCalled()
  })

  it('should be called password change', () => {
    wrapper.instance().handleChange = jest.fn()
    wrapper.find('.password').simulate('change')
    expect(wrapper.instance().handleChange).toHaveBeenCalled()
  })

  it('should set state', () => {
    expect(wrapper.state('email')).toEqual('')
    wrapper.instance().handleChange(mockEvent)
    expect(wrapper.state('email')).toEqual('nathan.froeh@gmail.com')
  })

  



})