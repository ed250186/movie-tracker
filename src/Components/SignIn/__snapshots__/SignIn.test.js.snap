// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn should match snapshot 1`] = `
<div
  className="sign-in"
>
  <NavLink
    className="exit"
    to="/"
  />
  <form
    className="form"
    onSubmit={[Function]}
  >
    <NavLink
      to="/"
    >
      <img
        alt="exit sign-in button"
        className="exit-button"
        src="cancel.png"
      />
    </NavLink>
    <div>
      <input
        className="email"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        value=""
      />
      <input
        className="password"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
      <input
        className="button"
        type="submit"
        value="Sign In"
      />
      <p />
    </div>
  </form>
</div>
`;
